lib/java/com/google/android/material/datepicker/MaterialDatePicker.java

```diff
@@ -262,6 +272,32 @@ public void onClick(View v) {
             dismiss();
           }
         });
+
+    clearButton = root.findViewById(R.id.clear_button);
+    if (clearable) {
+      if (getDateSelector().isSelectionComplete()) {
+        clearButton.setEnabled(true);
+      } else {
+        clearButton.setEnabled(false);
+      }
+      clearButton.setTag(CLEAR_BUTTON_TAG);
+      clearButton.setOnClickListener(
+              new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                  for (View.OnClickListener listener : onClearButtonClickListeners) {
+                    listener.onClick(v);
+                  }
+                  dismiss();
```

Comment:
Seems not correct. Shouldn't we clear the selection here instead of dismiss the date picker?